docs/integrating-with-relay.md

```diff
@@ -0,0 +1,179 @@
+
+Steps have a relatively easy lot in life. They run, do some work to advance the state of the workflow, and exit. In many cases, a step can use existing scripts which you modify only enough to take advantage of the Relay service API. The [Getting started](./getting-started.md) shows an example workflow that uses the `input` map to access this API for retrieving and setting keys and values to pass data through the workflow. For more advanced uses, you'll probably need a script that uses the [ni utility](./cli/ni.md) or its [python SDK equivalent](https://github.com/puppetlabs/nebula-sdk/blob/master/support/python/src/nebula_sdk/interface.py) to get and set data. Formally, a step container running in Relay can expect:
+
+* It will be executed without persistent storage
+* Only keys declared in the step's `spec` map in the workflow will be available in the metadata service
+* It can set output values through the metadata service which will be available to later steps
+* Exiting with a zero exit code will cause the workflow run to continue
+* Exiting with a non-zero exit code will cause the workflow run to be terminated and no dependent steps will run
+
+For examples of using the `ni` utility in shell commands, see the [kustomize integration](https://github.com/relay-integrations/relay-kustomize/blob/master/actions/steps/kustomize/step.sh); for examples of using the Python SDK, the [AWS EC2 integration](https://github.com/relay-integrations/relay-aws-ec2/tree/master/actions/steps) has several steps of varying complexity.
+
+### Building and publishing actions
+
+Once you've got your base image and custom code together, you can proceed with building and publishing the action containers. Relay has conventions for container and integration metadata which aren't _required_ but will increase consistency and help with future compatibility. These conventions are encoded in the [template integration repo](https://github.com/relay-integrations/template), specifically:
+
+* Integration repos should have an `integration.yaml` at their root, whose structure is defined in the [Integration RFC](TODO:link).
+* Relay containers should be built with LABEL directives that indicate their title and description. 
+
+The following Dockerfile shows this in action:
+
+```shell
+FROM relaysh:core-python
+COPY "./handler.py" "/step-github-trigger-pull-request-merged.py"
+ENTRYPOINT []
+CMD ["python3", "/step-github-trigger-pull-request-merged.py"]
+
+LABEL "org.opencontainers.image.title"="GitHub pull request merged"
+LABEL "org.opencontainers.image.description"="This trigger fires when a GitHub pull request is merged."
+```
+
+You can publish images to any publicly-accessible container registry. The service defaults to Docker Hub for workflow `image` attributes that are not fully-qualified with a hostname, so these are equivalent:
+
+```yaml
+steps:
+  - name: default-registry
+    image: relaysh/core
+  - name: explicit-registry
+    image: hub.docker.com/relaysh/core
```

Comment:
I think it's actually (weirdly) `index.docker.io` - did you verify this works as-is?

Reply:
good catch, fixed